File: app/src/main/java/zone/com/zrefreshlayoutdemo/SinaHeaderActivity.java
package zone.com.zrefreshlayoutdemo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;

import and.utils.activity_fragment_ui.ToastUtils;
import butterknife.Bind;
import butterknife.ButterKnife;
import zone.com.zrefreshlayout.ZRefreshLayout;
import zone.com.zrefreshlayout.footer.LoadFooter;
import zone.com.zrefreshlayout.header.SinaRefreshHeader;

/**
 * Created by fuzhipeng on 2017/1/11.
 */

public class SinaHeaderActivity extends AppCompatActivity {
    @Bind(R.id.iv)
    ImageView iv;
    @Bind(R.id.refresh)
    ZRefreshLayout refresh;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.auto_refresh);
        ButterKnife.bind(this);
        refresh.setIHeaderView(new SinaRefreshHeader());
        refresh.setPullListener(new ZRefreshLayout.PullListener() {
            @Override
            public void refresh(final ZRefreshLayout zRefreshLayout) {
                refresh.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        iv.setImageResource(R.drawable.aaaaaaaaaaaab);
                        zRefreshLayout.refreshComplete();
                    }
                }, 2000);
            }

            @Override
            public void complete(ZRefreshLayout zRefreshLayout) {

            }
        });

        refresh.setIFooterView(new LoadFooter());
        refresh.setLoadMoreListener(new ZRefreshLayout.LoadMoreListener() {
            @Override
            public void loadMore(final ZRefreshLayout zRefreshLayout) {
                refresh.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        ToastUtils.showShort(SinaHeaderActivity.this,"加载更多");
                        zRefreshLayout.loadMoreComplete();
                    }
                }, 2000);
            }

            @Override
            public void complete(ZRefreshLayout zRefreshLayout) {

            }
        });
    }
}

File: app/src/main/java/zone/com/zrefreshlayoutdemo/SinaHeaderActivity.kt
package zone.com.zrefreshlayoutdemo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

import com.zone.lib.utils.activity_fragment_ui.ToastUtils

import kotlinx.android.synthetic.main.auto_refresh.*
import zone.com.zrefreshlayout.ZRefreshLayout
import zone.com.zrefreshlayout.footer.LoadFooter
import zone.com.zrefreshlayout.header.SinaRefreshHeader

/**
 * Created by fuzhipeng on 2017/1/11.
 */

class SinaHeaderActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.auto_refresh)
        refresh.iHeaderView = SinaRefreshHeader()
        refresh.pullListener = ZRefreshLayout.PullListener { zRefreshLayout ->
            refresh.postDelayed({
                iv.setImageResource(R.drawable.aaaaaaaaaaaab)
                zRefreshLayout.refreshComplete()
            }, 2000)
        }

        refresh.iFooterView = LoadFooter()
        refresh.loadMoreListener = ZRefreshLayout.LoadMoreListener { zRefreshLayout ->
            refresh.postDelayed({
                ToastUtils.showShort(this@SinaHeaderActivity, "加载更多")
                zRefreshLayout.loadMoreComplete()
            }, 2000)
        }
    }
}
